src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs

```diff
@@ -11,7 +11,14 @@ public class FailOverNamespacePartitioning : INamespacePartitioningStrategy
 
         public FailOverNamespacePartitioning(ReadOnlySettings settings)
         {
-            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces, out namespaces) || namespaces.Count != 2)
+            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Partitioning.Namespaces, out namespaces))
+            {
+                throw new ConfigurationErrorsException("The 'FailOver' namespace partitioning strategy requires exactly two namespaces to be configured");
+            }
+
+            namespaces = new NamespaceConfigurations(namespaces.Where(n => n.Purpose == NamespacePurpose.Partitioning).ToList());
+
+            if (namespaces.Count != 2)
```

Comment:
Namespaces are registered, but not for the right purpose. Should the 2nd exception indicate so to guide the users?